frontend/views/containers/contributions/IncomeDetails.vue

```diff
@@ -0,0 +1,248 @@
+import Tooltip from '@components/Tooltip.vue'
+import ModalBaseTemplate from '@components/Modal/ModalBaseTemplate.vue'
+import TextWho from '@components/TextWho.vue'
+import GroupPledgesGraph from '../GroupPledgesGraph.vue'
+import L from '@view-utils/translations.js'
+
+export default {
+  name: 'IncomeDetails',
+  mixins: [validationMixin],
+  components: {
+    ModalBaseTemplate,
+    InputAmount,
+    TextWho,
+    Tooltip,
+    PaymentMethods,
+    GroupPledgesGraph
+  },
+  data () {
+    return {
+      ephemeral: {
+        formFeedbackMsg: {}
+      },
+      form: {
+        incomeDetailsKey: null,
+        amount: null
+      },
+      fakeStore: {
+        incomeDetailsKey: 'pledgeAmount', // incomeAmount || pledgeAmount
+        amount: 100
+      }
+    }
+  },
+  computed: {
+    ...mapGetters([
+      'groupMembers',
+      'groupSettings',
+      'memberProfile',
+      'groupMincomeFormatted',
+      'groupMincomeSymbolWithCode',
+      'ourUsername',
+      'groupMembersByUsername'
+    ]),
+    memberGroupProfile () {
+      const profile = this.memberProfile(this.ourUsername) || {}
+      return profile.groupProfile || {}
+    },
+    needsIncome () {
+      return this.form.incomeDetailsKey === 'incomeAmount'
+    },
+    whoIsPledging () {
+      return this.groupMembersByUsername.filter(username => this.groupMembers[username].groupProfile.incomeDetailsKey === 'pledgeAmount' && username !== this.ourUsername)
+    }
+  },
+  created () {
+    const incomeDetailsKey = this.memberGroupProfile.incomeDetailsKey
+    if (incomeDetailsKey) {
+      this.form.incomeDetailsKey = incomeDetailsKey
+      this.form.amount = this.memberGroupProfile[incomeDetailsKey]
+    }
+  },
+  methods: {
+    resetAmount () {
+      this.form.amount = this.form.incomeDetailsKey === this.memberGroupProfile.incomeDetailsKey ? this.memberGroupProfile[this.memberGroupProfile.incomeDetailsKey] : ''
+      this.$v.form.$reset()
+    },
+    closeModal () {
+      this.$refs.modal.close()
+    },
+    async submit () {
+      if (this.$v.form.$invalid) {
+        alert(L('Invalid payment info'))
+        return
+      }
+      try {
+        const incomeDetailsKey = this.form.incomeDetailsKey
+        const groupProfileUpdate = await sbp('gi.contracts/group/groupProfileUpdate/create',
+          {
+            incomeDetailsKey,
+            [incomeDetailsKey]: +this.form.amount
+          },
+          this.$store.state.currentGroupId
+        )
+        await sbp('backend/publishLogEntry', groupProfileUpdate)
+        this.closeModal()
+      } catch (e) {
+        console.error('setPaymentInfo', e)
+        this.ephemeral.formFeedbackMsg = {
+          text: L(`Failed to update income details. Error: ${e.message}`),
+          class: 'has-text-danger'
+        }
+      }
+    }
+  },
+  validations: {
+    form: {
+      incomeDetailsKey: {
+        [L('This field is required')]: required
+      },
+      amount: {
```

Comment:
This needs 2 validations, like in Contributions.vue, one named `incomeAmount` and the other named `pledgeAmount`.

It's indeed a good thing that we're keeping them separate. This makes it possible for you to easily write different validations for them as you're doing here. The `pledgeAmount` validation, for example, does not need to make sure the pledge is lower than the mincome.